cfme/cloud/keypairs.py

```diff
@@ -154,51 +197,21 @@ def delete(self, cancel=False, wait=False):
             self.wait_for_delete()
 
     def wait_for_delete(self):
-        def refresh():
-            self.provider.refresh_provider_relationships()
-            view.browser.selenium.refresh()
-            view.flush_widget_cache()
-        view = navigate_to(self, 'All')
+        self.provider.refresh_provider_relationships()
         # look for the row, call is_displayed on it to get boolean
         # find_row_on_pages is going to raise NoSuchElement when the row disappears
         try:
-            wait_for(lambda: view.paginator.find_row_on_pages(view.entities.table,
-                                                              name=self.name).is_displayed,
-                     fail_condition=True, message="Wait keypairs to disappear", num_sec=500,
-                     delay=5, fail_func=refresh)
+            wait_for(
```

Comment:
This is causing problems for me in testing.

The 'refresh' method exists because a browser refresh or a provider refresh alone isn't sufficient. I'm seeing fast timeouts when calling delete that suggest that the fail_condition check isn't working like you intend.

Also you're passing num_sec and timeout, which are the same.

Reply:
Really this entire `wait_for_delete()` method should be removed anyway.

Reply:
@mshriver if I remove `wait_for_delete()` method then `self.exists` will return error after it tries to navigate to keypair details page and assert will fail.

Reply:
The `exists` method for any object should return Boolean, not raise an error, unless something unexpected happens. For example we'll know if we try to navigate and it doesn't exist we'll get `CandidateNotFound` when trying to navigate using accordion - so the `exists` method catches CandidateNotFound and returns false. If some _other_ exception happens, then it won't be caught.

Reply:
yes that is true, but after initiate delete keypair took some time to disappear from UI.
`wait_for_delete()` is required there.